integration/container/links_linux_test.go

```diff
@@ -29,21 +25,13 @@ func TestLinksEtcHostsContentMatch(t *testing.T) {
 	client := request.NewAPIClient(t)
 	ctx := context.Background()
 
-	cID := container.Run(t, ctx, client, container.WithCmd("cat", "/etc/hosts"), container.WithNetworkMode("host"))
-
-	poll.WaitOn(t, container.IsStopped(ctx, client, cID), poll.WithDelay(100*time.Millisecond))
-
-	body, err := client.ContainerLogs(ctx, cID, types.ContainerLogsOptions{
-		ShowStdout: true,
-	})
-	require.NoError(t, err)
-	defer body.Close()
-
-	var b bytes.Buffer
-	_, err = stdcopy.StdCopy(&b, ioutil.Discard, body)
+	cID := container.Run(t, ctx, client, container.WithNetworkMode("host"))
```

Comment:
Maybe we could add
```
poll.WaitOn(t, container.IsInState(ctx, client, cID, "running"), poll.WithDelay(100*time.Millisecond))
```
so that to make sure the container is running?

Reply:
It looks like such a check is redundant.

As far as I can see from the source code, container.Run() is synchronous operation, i.e. a successful return from container.Run() means the container is running (and its status is set to running)

Surely, a container can exit in between calls to container.Run() and container.Exec(), but there's no need to check for that either, as this is done in Exec().